### PR TITLE
allow to bring your own assert and memory functions

### LINKED ISSUE #5: assert replacement
It would be nice to be able to bring my own assert like stb headers allow. Maybe something like `vxl_assert` instead of plain `assert`.

### PR DESCRIPTION
**Pull request by mgerhardy**
fixes #5

### CODE REVIEW

File: libvxl.c [hunk 1]
```diff
@@ -461,7 +468,7 @@ void libvxl_write(struct libvxl_map* map, void* out, size_t* size) {
 	size_t sx = (map->width + LIBVXL_CHUNK_SIZE - 1) / LIBVXL_CHUNK_SIZE;
 	size_t sy = (map->height + LIBVXL_CHUNK_SIZE - 1) / LIBVXL_CHUNK_SIZE;
 
-	size_t *chunk_offsets = (size_t*)libvxl_mem_malloc(sx * sy);
```
Comment (xtreme8000) [inline libvxl.c]: I also fixed this error

**Comment by mgerhardy** [comment]
Sorry that I’ve made the changes in both PRs - that wasn’t intended. The memory allocation for the dynamic sized array is needed for MSVC because it’s a C99 feature and MSVC is way behind when it comes down to the C standard

**Comment by mgerhardy** [comment]
Thanks for merging

**Comment by xtreme8000** [comment]
No worries, the VLA was way to large (1MiB usually) for the stack anyway, so heap is better.
edit: nvm didn't see the division by LIBVXL_CHUNK_SIZE (4KiB in that case)

**Pull request merged**